artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java

```diff
@@ -306,7 +306,7 @@ public int handleDeliver(MessageReference reference, ICoreMessage message) {
          int size = dispatch.getMessage().getSize();
          reference.setProtocolData(MessageId.class, dispatch.getMessage().getMessageId());
          session.deliverMessage(dispatch);
-         currentWindow.decrementAndGet();
+         currentWindow.updateAndGet(i -> i > 0 ? i - 1 : i);
```

Comment:
This change causes the failure of org.apache.activemq.artemis.tests.integration.stomp.v12.StompV12Test.testSubscribeWithClientAckThenConsumingAgainWithAutoAckWithNoDisconnectFrame

Reply:
Ignore my comment, I can reproduce this failure also on the current main branch. I need to deepen.

Reply:
The test is a STOMP test that only uses the core client and doesn't cross into this changed openwire code so must be unrelated.